apps/emqx_connector/src/emqx_connector_redis.erl

```diff
@@ -185,24 +171,10 @@ redis_fields() ->
     , {auto_reconnect, fun emqx_connector_schema_lib:auto_reconnect/1}
     ].
 
-to_server(Server) ->
-    try {ok, parse_server(Server)}
-    catch
-        throw : Error  ->
-            Error
-    end.
-
-to_servers(Servers) ->
-    try {ok, lists:map(fun parse_server/1, string:tokens(Servers, ", "))}
-    catch
-        throw : _Reason ->
-            {error, Servers}
-    end.
+server(type) -> binary();
+server(validator) -> [?NOT_EMPTY("the value of the field 'server' cannot be empty")];
```

Comment:
I'm not sure, since it doesn't seem to parse the input string into that tuple :thinking: 

Also, the failing checks in CI suggest that maybe it expects this field to contain something different, like this output from `typerefl`:

```erlang
1> emqx_connector_redis:server().
{'$type_refl',#{check => fun erlang:is_tuple/1,
                definition =>
                    [{["emqx_connector_redis:server","(",[],")"],"tuple()"}],
                from_string => fun emqx_connector_redis:to_server/1,
                name => "emqx_connector_redis:server()"}}
```

Reply:
```
server(type) -> {binary(), integer()};
```
The original code specified a typerfl `server()` type which had information about how to parse binaries into `{binary(), integer()}` tuples or their lists (`from_string` key in @thalesmg's output).

Reply:
Yes...the problem is coming from the auth apps which is why I didn't notice it would cause an issue as I was running the connector app tests only when doing the changes. Though the previous version, IIRC, passed without the forced change of the strings into tuples, which at first glance seems like it should have potentially caused an issue as well as eredis seems to expect tuples when being started.

Within eredis there is this line:
`init_nodes = [#node{address= A, port = P} || {A,P} <- proplists:get_value(servers, Opts, [])],`

That fails, by returning an empty list, if the list of servers is not tuples. Which should lead to this error condition being met:
```
init([PoolName, Opts]) ->
    process_flag(trap_exit, true),
    case connect_(PoolName, Opts) of
        #state{init_nodes = []} ->
            {stop, <<"ERR unable to connect to any nodes">>};
```

Without eredis being started/connected properly I would think the auth tests should have failed.

...10 minutes later...

And here is why they didn't fail:

In the auth tests things are started like this:
```
            {ok, _} = emqx_resource:create_local(
              ?REDIS_RESOURCE,
              emqx_connector_redis,
              redis_config()),

...

redis_config() ->
    #{auto_reconnect => true,
      database => 1,
      pool_size => 8,
      redis_type => single,
      password => "public",
      server => {?REDIS_HOST, ?REDIS_PORT}, <======= server string "pre broken"
      ssl => #{enable => false}
     }.
```

So when I had the server definition as a binary it passed whatever checks are failing now, and passed the auth tests because the resource was created directly/manually not via the strings like are expected to be defined.

And now when I say it has to be a binary, but I provide no mapping option to swap a string into a tuple it fails. Seems I should probably just swap the whole redis module back to the way it was...though the reason I'd actually done the change was because in all other cases I've dealt with anything that comes with a type either comes with a default or with the option for the value to be nullable. The redis module breaks this expectation. Need to figure out what to do about that.